### PR TITLE
[Chore] Test api endpoints [WIP]

### PR DESCRIPTION
**Pull request by JoshuaOndieki**
**What does this PR do?**
Adds tests for the API endpoints

**Endpoints being tested?**

EndPoint | Functionality
-- | --
POST `/api/v1/rides` | Creates a new ride
GET `/api/v1/rides` | Fetches all available rides
GET `/api/v1/rides/<rideId>` | Fetches a single ride
POST `/api/v1/auth/rides/<rideId>/requests` | Creates a ride request

**Tasks achieved in this PR**
- [x] Test ride creation edge cases
- [x] Test all rides fetching edge cases
- [x] Test single ride fetching edge cases
- [x] Test ride request edge cases

**How should this be manually tested?**
- Clone this repo using `git clone https://github.com/JoshuaOndieki/ridemyway-api.git`
- Check the ReadMe for instructions on setting up the project.
- `git checkout ch-test-api-endpoints` 
- Run `nosetests  --with-coverage --cover-package=ridemyway` to see the tests coverage

**Background Context?**
The tests are failing since the functionality is yet to be implemented.
My approach assumes no users and therefore rides can be created and requests made without user interaction.

**Resources & Libs**
[This Pivotal Tracker board](https://www.pivotaltracker.com/n/projects/2179581) was used to plan and manage the tasks.

### CODE REVIEW

File: ridemyway/tests/test_create_ride.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+            'data':
+                [{'departure': 'Not a date',
+                  'origin': 'Nairobi',
+                  'destination': 'Garissa',
+                  'cost': '%^$', 'vehicle_number_plate':
+                  'KBC-A21', 'capacity': 'hundred'}]}
+
+        self.response = self.client().post('/api/v1/rides', data=data)
+        result = json.loads(self.response.data.decode())
+        self.assertEqual(result['message'], "Invalid data given")
```
Comment (meshack-mbuvi) [inline ridemyway/tests/test_create_ride.py]: what happens if I a pass *vehicle_number_plate* as numbers instead of strings.
Think of that edge case.

File: ridemyway/tests/test_fetch_rides.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        self.assertEqual(result['message'], 'Rides retrieved successfully')
+        self.assertEqual(self.response.status_code, 200)
+
+    def test_fetches_a_single_ride_with_valid_id(self):
+        self.response = self.client().get('/api/v1/rides/1')
+        result = json.loads(self.response.data.decode())
+        self.assertEqual(result['message'], 'Ride retrieved successfully')
+        self.assertEqual(self.response.status_code, 200)
+
+    def test_does_not_fetch_single_ride_that_does_not_exist(self):
```
Comment (meshack-mbuvi) [inline ridemyway/tests/test_fetch_rides.py]: suggested edit:
change **test_does_not_fetch_single_ride_that_does_not_exist** to **test_does_not_fetch_ride_that_does_not_exist**

File: ridemyway/tests/test_ride_request.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+        result = json.loads(self.response.data.decode())
+        self.assertEqual(result['message'], 'Ride request created successfully')
+        self.assertEqual(self.response.status_code, 201)
+
+    def test_does_not_create_a_request_to_nonexistent_ride(self):
+        self.response = self.client().get('/api/v1/rides/6454')
+        result = json.loads(self.response.data.decode())
+        self.assertEqual(result['status'], 'failed')
+        self.assertEqual(self.response.status_code, 404)
+
```
Comment (meshack-mbuvi) [inline ridemyway/tests/test_ride_request.py]: suggested missing edge case.: **test_does_not_create_request_for_expired_ride**
*NB:* Expired in terms of date and time

Comment (JoshuaOndieki) [reply ridemyway/tests/test_ride_request.py]: How do I test that, if at the same time I can not add a ride with a past date?

**Comment by TheSteelGuy** [comment]
quite okay i did not see anything alarming.

**Comment by JoshuaOndieki** [comment]
@meshack-mbuvi @TheSteelGuy  Thanks for the feedback.
@meshack-mbuvi, Lemme know how I can test for expired rides

**Comment by larryTheGeek** [comment]
Good Job on The PR. Very detailed PR

**Comment by meshack-mbuvi** [comment]
For my case,have compared date stored with current timestamp

**Comment by JoshuaOndieki** [comment]
I have no idea how I will put an expired ride into my app and my create ride should not allow passed dates.
Unless I create a ride and then wait for it to expire then try creating a request.
The downside is that it will slow my tests
